influxdb_iox/tests/end_to_end_cases/management_api.rs

```diff
@@ -1949,3 +1949,175 @@ async fn test_compact_os_chunks() {
     assert_ne!(new_chunk_id, chunk_id_1);
     assert_ne!(new_chunk_id, chunk_id_2);
 }
+
+#[tokio::test]
+async fn test_compact_os_partition() {
+    use data_types::chunk_metadata::ChunkStorage;
+
+    let fixture = ServerFixture::create_shared(ServerType::Database).await;
+    let mut write_client = fixture.write_client();
+    let mut management_client = fixture.management_client();
+    let mut operations_client = fixture.operations_client();
+
+    let db_name = rand_name();
+    DatabaseBuilder::new(db_name.clone())
+        .persist(true)
+        .persist_age_threshold_seconds(1_000)
+        .late_arrive_window_seconds(1)
+        .build(fixture.grpc_channel())
+        .await;
+
+    // Chunk 1
+    let lp_lines = vec!["cpu,tag1=cupcakes bar=1 10", "cpu,tag1=cookies bar=2 10"];
+
+    let num_lines_written = write_client
+        .write_lp(&db_name, lp_lines.join("\n"), 0)
+        .await
+        .expect("write succeded");
+    assert_eq!(num_lines_written, 2);
+
+    wait_for_exact_chunk_states(
+        &fixture,
+        &db_name,
+        vec![ChunkStorage::OpenMutableBuffer],
+        std::time::Duration::from_secs(5),
+    )
+    .await;
+
+    let chunks = management_client
+        .list_chunks(&db_name)
+        .await
+        .expect("listing chunks");
+    assert_eq!(chunks.len(), 1);
+    let partition_key = &chunks[0].partition_key;
+
+    management_client
+        .persist_partition(&db_name, "cpu", &partition_key[..], true)
```

Comment:
```suggestion
        .persist_partition(&db_name, "cpu", partition_key.as_ref(), true)
```

works as well, whatever you prefer.